src/app/plugins/platform/Docker/plugins/cms/Drupal8/installDrupal.ts

```diff
@@ -36,82 +51,61 @@ export interface InstallDrupalOptions {
-  // This will help prevent a lot of headaches due to misalignment with the platform's
-  // requirements.
-  if (projectType === pantheonProject && needsRename) {
-    throw new Error(
-      `Pantheon projects do not support '${documentRoot}' as the document root.`,
-    );
-  }
+/**
+ * Installs Drupal from a given project installation type.
+ *
+ * @param {InstallDrupalOptions}
+ * @returns {Promise<void>}
+ */
+async function installDrupal(
+  installOptions: InstallDrupalOptions,
+): Promise<void> {
+  // @todo Revise this to use better syntax options.
+  const projectType = installOptions.projectType;
 
-  await makeDir(serviceDirectory);
+  // Confirm a valid web root was selected for the platform.
+  // @todo Refactor for cleaner use of params.
+  await validateWebRoot(installOptions);
 
-  await spawnComposer(
-    [
-      'create-project',
-      projectType,
-      'drupal',
-      '--stability',
-      'dev',
-      '--no-interaction',
-      '--ignore-platform-reqs',
-    ],
-    { cwd: serviceDirectory },
-  );
+  // Prepare the service container directory for Drupal.
+  await scaffoldServiceDirectory(installOptions);
 
-  const drupalRoot = path.join(serviceDirectory, 'drupal');
+  // Ensure the web root is renamed if necessary.
+  const drupalRoot = await renameWebRoot(installOptions);
 
-  // Rename 'web' in generated files, if we need to
-  if (needsRename) {
-    await Promise.all([
-      replaceIn(
-        path.join(drupalRoot, 'composer.json'),
-        /web\//g,
-        documentRoot + '/',
-      ),
-      replaceIn(
-        path.join(drupalRoot, '.gitignore'),
-        /web\//g,
-        documentRoot + '/',
-      ),
-      replaceIn(
-        path.join(drupalRoot, 'scripts/composer/ScriptHandler.php'),
-        /\/web/g,
-        '/' + documentRoot,
-      ),
-      replaceIn(path.join(drupalRoot, 'README.md'), /web/g, documentRoot),
-    ]);
-  }
+  // Apply post-scaffolding composer updates.
+  await applyComposerUpdates(drupalRoot);
 
-  // Composer create-project now "helpfully" installs the web/ directory for us, which
-  // we don't want when it isn't the document root.
-  if (needsRename) {
-    const webRoot = path.join(drupalRoot, 'web');
-    await rimrafAsync(webRoot);
-  }
+  // Perform project-specific operations.
+  await customizeByProjectType(projectType, drupalRoot);
+}
 
+/**
+ * Apply composer updates needed after initial scaffolding.
+ *
+ * @param {string} drupalRoot
+ */
+async function applyComposerUpdates(drupalRoot: string) {
   // Make sure the lock file is up to date.
   await spawnComposer(['update', '--lock', '--ignore-platform-reqs'], {
     cwd: drupalRoot,
   });
+}
 
-  // Perform project-specific operations.
+/**
+ * Apply type-specific project customizations as needed.
+ *
+ * @param {string} projectType
+ * @param {string} drupalRoot
+ */
+async function customizeByProjectType(projectType: string, drupalRoot: string) {
```

Comment:
Prefer the stricter `ProjectType` type here - by allowing `projectType` to be an arbitrary string, you're letting people accidentally call `customizeByProjectType(drupalRoot, projectType)` which is a typo that can be caught at compile time (or earlier, if your editor understands TypeScript).